deltachat-ffi/src/lib.rs

```diff
@@ -2451,7 +2453,14 @@ pub unsafe extern "C" fn dc_chat_get_archived(chat: *mut dc_chat_t) -> libc::c_i
         return 0;
     }
     let ffi_chat = &*chat;
-    ffi_chat.chat.is_archived() as libc::c_int
+    let ffi_context = &*ffi_chat.context;
+    ffi_context
+        .with_inner(|ctx| match ffi_chat.chat.get_id().get_archive_state(ctx) {
+            ArchiveState::Normal => 0,
```

Comment:
this sholud be `state.into()` using an implementation of `From<ArchiveState> for usize`